src/kartograf/atom_mapper.py

```diff
@@ -873,10 +910,37 @@ def suggest_mappings(
         Iterator[AtomMapping]
             returns an interator of possible atom mappings.
         """
-        yield LigandAtomMapping(
-            A,
-            B,
-            self.suggest_mapping_from_rdmols(
-                molA=A.to_rdkit(), molB=B.to_rdkit()
-            ),
-        )
+
+        if isinstance(A, ProteinComponent) or isinstance(B, ProteinComponent):
```

Comment:
Rather than splitting specifically over proteincomponent, could we use rdkit's GetMolFrags to check for discontinuouus molecules and then map each continuous molecule and then merge them?

Reply:
1. This is a good idea, one could skip the openMM top like this. In anyway I think the current implementation works for now ans we should change it to the rdkit frags approach before merging. (Can do that when I'm back, If that works with the timelines.

2. One thing, I m just trying to think about, do we want to ensure protein protein mapping as a fragment case, or could there be a component agnostic approach or other components combinations? (Nucleotide double strands????) But what does it mean for small molecules 🤔

Reply:
The general take is that things should be as component agnostic as possible - protein component is a thing today but it's very likely to not be a thing in gufe 2.0, etc...

Reply:
That's a good idea, one caveat I found with this approach is that if there are waters then every water becomes a new molecule, which could be fine, but I think it probably deserves more thinking about this. That said, the current approach doesn't seem to handle that case correctly either way, for some reason it doesn't put all the atoms of the waters in the second protein component when splitting using chains.

Reply:
I think this has been already resolved, we are using mol frags.